packages/server/src/utilities/scriptRunner.ts

```diff
@@ -1,62 +1,111 @@
 import ivm from "isolated-vm"
 
+import bson from "bson"
+import { BundleType, loadBundle } from "../jsRunner/bundles"
+import env from "../environment"
+
 const JS_TIMEOUT_MS = 1000
 
 class ScriptRunner {
   vm: IsolatedVM
 
-  constructor(script: string, context: any) {
-    const code = `let fn = () => {\n${script}\n}; results.out = fn();`
-    this.vm = new IsolatedVM({ memoryLimit: 8 })
-    this.vm.context = {
-      ...context,
-      results: { out: "" },
+  constructor(script: string, context: any, { parseBson = false } = {}) {
+    this.vm = new IsolatedVM({
+      memoryLimit: env.JS_RUNNER_MEMORY_LIMIT,
+      parseBson,
+      context: {
+        ...context,
+        data: parseBson
+          ? bson.BSON.serialize({ data: context.data })
+          : context.data,
+      },
+    })
+
+    if (parseBson) {
+      script = `return JSON.parse(JSON.stringify((function(){data=deserialize(data).data;${script}})()));`
     }
-    this.vm.code = code
+
+    this.vm.code = script
   }
 
   execute() {
     this.vm.runScript()
-    const results = this.vm.getValue("results")
-    return results.out
+    const result = this.vm.getResult()
```

Comment:
NAB: could make this inline: `return this.vm.getResult()`

Reply:
On build time this will inlined, and having it as separate lines makes debugging much easier